### PR TITLE
json views to restful api

### PR DESCRIPTION
**Pull request by casey1173**
Replaced legacy django json views with django restful framework. The main reason to do this is to leverage consistency and use the same patterns for the same type of code. 

```
There should be one-- and preferably only one --obvious way to do it.
```

https://www.python.org/dev/peps/pep-0020/

Bonus:
- hotfix for generational form display choices
- hotfix for dose activation on endpoints
- fix untagged references layout issue

### CODE REVIEW

**Comment by casey1173** [comment]
https://github.com/shapiromatron/hawc/blob/397fb3c4d4d532364d2a9392dd58abbc6cb73d17/hawc/apps/lit/views.py#L252-L256
should this one be updated too?

**Comment by shapiromatron** [comment]
@casey1173 If I recall that one's pretty complicated. It's fine to skip this one, it's a bit of a God (anti)pattern that I can investigate untangling.

File: hawc/apps/lit/api.py [hunk 1]
```diff
@@ -93,6 +112,8 @@ def references(self, request, pk):
             .order_by("id")
         )
         serializer = serializers.ReferenceSerializer(page, many=True)
+
+        return Response(serializer.data)
         return self.get_paginated_response(serializer.data)
```
[the remark this answers is not in the document]
Comment (shapiromatron) [reply hawc/apps/lit/api.py:117]: we might need to have two different endpoints `/references/` for pagination and `/references/all/` for no pagination

**Review by shapiromatron — CHANGES_REQUESTED** [review]
Looking great so far. A few minor edits, mostly leveraging some of the built-in djago-rest framework instead of our custom code.

**Review by shapiromatron — APPROVED** [review]
Looks great. I went overboard and kept fixing other things and legacy views that I saw when I reviewed your code.    The end result is i think we have a much more readable code-base and more standardization on where things should go.

Nice job!

**Pull request merged**